src/github.com/matrix-org/dendron/proxy/proxy.go

```diff
@@ -105,3 +110,21 @@ func (p *SynapseProxy) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 
 	p.ProxyHTTP(w, req.Method, req.URL, req.Body, req.ContentLength, req.Header)
 }
+
+func (p *SynapseProxy) measure(req *http.Request) (*http.Response, error) {
+	start := time.Now()
+	resp, httpErr := p.Client.Do(req)
+
+	if p.Metrics != nil {
+		metric, err := p.Metrics.GetMetricWithLabelValues(req.URL.Path, req.Method)
```

Comment:
bleh, yeah, that will blow up a fair bit. I really wish those things weren't in paths.

In the mean time, I have added heuristic replacement for them